x-pack/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/package_installation_and_upgrade.md

```diff
@@ -0,0 +1,165 @@
+```
+
+#### **Scenario: Package is installed via bundled Fleet package in Kibana**
+
+**Automation**: 2 integration tests.
+
+```Gherkin
+Given the package is not installed
+And user is in an air-gapped environment
+When user opens the Rule Management page
+Then the package gets installed in the background from packages bundled into Kibana
+```
+
+#### **Scenario: Large package can be installed on a small Kibana instance**
+
+**Automation**: 1 integration test.
+
+```Gherkin
+Given the package is not installed
+And the package contains the largest amount of historical rule versions (15000)
+And the Kibana instance has a memory heap size of 700 Mb (see note below)
+When user opens the Rule Management page
+Then the package is installed without Kibana crashing with an Out Of Memory error
+```
+
+**Note**: 600 Mb seems to always crash Kibana with an OOM error. 700 Mb runs with no issues in the Flaky test runner with 100 iterations: https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/2215.
+
+### Scenarios for the real package
+
+#### **Scenario: User can install prebuilt rules from scratch, then install new rules and upgrade existing rules from the new package**
+
+**Automation**: 1 integration test with real packages.
+
+```Gherkin
+Given there are two package versions: N-1 and N
+And the package of N-1 version is installed
+When user calls the status endpoint
+Then it should return a 200 response with some number of rules to install and 0 rules to upgrade
+When user calls the installation/_review endpoint
+Then it should return a 200 response matching the response of the status endpoint
+When user calls the installation/_perform_ endpoint
+Then it should return a 200 response matching the response of the status endpoint
+And rules returned in this response should exist as alert saved objects
+When user installs the package of N version
+Then it should be installed successfully
+When user calls the status endpoint
+Then it should return a 200 response with some number of new rules to install and some number of rules to upgrade
+When user calls the installation/_review endpoint
+Then it should return a 200 response matching the response of the status endpoint
+When user calls the installation/_perform_ endpoint
+Then rules returned in this response should exist as alert saved objects
+When user calls the upgrade/_review endpoint
+Then it should return a 200 response matching the response of the status endpoint
+When user calls the upgrade/_perform_ endpoint
+Then rules returned in this response should exist as alert saved objects
+```
+
+### Kibana upgrade
+
+#### **Scenario: User can use prebuilt rules after upgrading Kibana from version A to B**
+
+**Automation**: not automated, manual testing required.
+
+```Gherkin
+Given user is upgrading Kibana from version <A> to version <B>
+And the <A> instance contains already installed prebuilt rules
+When the upgrade is complete
+Then user should be able to install new prebuilt rules
```

Comment:
It'd be better to have two separate scenarios for that cases since set up differs. In case you still prefer to have one scenarios these two cases should be reflected in cases.